packages/flutter/test/cupertino/magnifier_test.dart

```diff
@@ -28,10 +28,8 @@ void main() {
               controller: magnifierController,
               magnifierInfo: magnifierInfo,
             ));
-
-    WidgetsBinding.instance.scheduleFrame();
-    await tester.pumpAndSettle();
-
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 2));
```

Comment:
Why are two pumps needed, and why 2 seconds?

Reply:
The first pump triggers the animations, the second ends them (two seconds is just "long enough to make sure all animations have finished"). This replaces "pumpAndSettle", which is generally a very sketchy way to do the same thing. ("pumpAndSettle" will fail to catch bugs where we require multiple frames to trigger or end an animation sequence. In general tests should never use "pumpAndSettle" unless they are verifying the return value.)

Reply:
Another thing to add to the styleguide? I have always tried to prefer pumpAndSettle because I thought it was actually more robust.

I feel like I'm confessing all my sins here on this review.

Reply:
It's already in the API docs for `pumpAndSettle` ("In general, it is better practice to figure out exactly why each frame is needed, and then to pump exactly as many frames as necessary. This will help catch regressions where, for instance, an animation is being started one frame later than it should."), but I'll add something to the style guide too. :-)